searchlib/src/vespa/searchlib/expression/functionnode.h

```diff
@@ -15,9 +14,9 @@ class FunctionNode : public ExpressionNode
     DECLARE_ABSTRACT_EXPRESSIONNODE(FunctionNode);
     const ResultNode * getResult() const override { return _tmpResult.get(); }
     ResultNode & updateResult() const { return *_tmpResult; }
-    virtual void reset() { _tmpResult.reset(NULL); }
+    virtual void reset() { _tmpResult.reset(nullptr); }
 
-    FunctionNode &setResult(const ResultNode::CP res) { _tmpResult = res; return *this; }
+    FunctionNode &setResult(const ResultNode::CP & res) { _tmpResult = res; return *this; }
```

Comment:
This change causes an extra clone() call

Reply:
Fixed